python/ray/serve/api.py

```diff
@@ -630,22 +638,12 @@ def start(
     if not ray.is_initialized():
         ray.init(namespace="serve")
 
-    current_namespace = ray.get_runtime_context().namespace
-    if detached:
-        if _UUID_RE.fullmatch(current_namespace) is not None:
-            raise RuntimeError(
-                "serve.start(detached=True) should not be called in anonymous "
-                "Ray namespaces because you won't be able to reconnect to the "
-                "Serve instance after the script exits. If you want to start "
-                "a long-lived Serve instance, provide a namespace when "
-                "connecting to Ray. See the documentation for more details: "
-                "https://docs.ray.io/en/master/namespaces.html?highlight=namespace#using-namespaces."  # noqa: E501
-            )
+    controller_namespace = _get_controller_namespace()
```

Comment:
Sorry, one last modification here: if `detached=False`, I think we should always use the current namespace and not default to the `serve` namespace. This would avoid two users on the same cluster stepping on each others' toes when using `detached=False` to start a temporary instance for testing/batch inference.

Reply:
Yes, sounds good - updated